src/NorthwindWebsite/Areas/Identity/Pages/Account/ExternalLogin.cshtml.cs

```diff
@@ -0,0 +1,208 @@
+                    externalLoginInfo.Principal.Identity.Name, 
+                    externalLoginInfo.LoginProvider);
+
+                return LocalRedirect(returnUrl);
+            }
+            if (result.IsLockedOut)
+            {
+                return RedirectToPage("./Lockout");
+            }
+
+            // If the user does not have an account, then ask the user to create an account.
+            ReturnUrl = returnUrl;
+            ProviderDisplayName = externalLoginInfo.ProviderDisplayName;
+            if (externalLoginInfo.Principal.HasClaim(claim => claim.Type == ClaimTypes.Email))
+            {
+                Input = new InputModel
+                {
+                    Email = externalLoginInfo.Principal.FindFirstValue(ClaimTypes.Email)
+                };
+            }
+
+            return Page();
+        }
+
+        public async Task<IActionResult> OnPostConfirmationAsync(string returnUrl = null)
+        {
+            returnUrl = returnUrl ?? Url.Content(UrlConstants.ReturnToHomepageUrl);
+            // Get the information about the user from the external login provider
+            var externalLoginInfo = await _signInManager.GetExternalLoginInfoAsync();
+            if (externalLoginInfo == null)
+            {
+                ErrorMessage = "Error loading external login information during confirmation.";
+                return RedirectToPage(UrlConstants.RedirectToLoginUrl, new { ReturnUrl = returnUrl });
+            }
+
+            if (ModelState.IsValid)
+            {
+                var user = CreateUser();
+
+                await _userStore.SetUserNameAsync(user, Input.Email, CancellationToken.None);
+                await _emailStore.SetEmailAsync(user, Input.Email, CancellationToken.None);
+
+                var result = await _userManager.CreateAsync(user);
+
+                if (!result.Succeeded)
+                {
+                    foreach (var error in result.Errors)
+                    {
+                        ModelState.AddModelError(string.Empty, error.Description);
+                    }
+                }
+
+                result = await _userManager.AddLoginAsync(user, externalLoginInfo);
+                if (result.Succeeded)
+                {
+                    _logger.LogInformation("User created an account using {Name} provider.", externalLoginInfo.LoginProvider);
+
+                    var userId = await _userManager.GetUserIdAsync(user);
+                    var code = await _userManager.GenerateEmailConfirmationTokenAsync(user);
+                    code = WebEncoders.Base64UrlEncode(Encoding.UTF8.GetBytes(code));
+                    var callbackUrl = Url.Page(
+                        "/Account/ConfirmEmail",
+                        pageHandler: null,
+                        values: new { area = "Identity", userId = userId, code = code },
+                        protocol: Request.Scheme);
+
+                    await Task.Run(() => _emailSender.SendEmailAsync(
```

Comment:
No sense to add `await` to `Task.Run`, in this way your task manager will pause method execution until `SendEmailAsync` will be completed. My point of view was to avoid this awaiting and just return user to page, because usage of basic email sender my require some long time to complete